parser/openfoodfacts_taxonomy_parser/parser/parser.py

```diff
@@ -234,14 +235,37 @@ def _write_to_database(self, taxonomy: Taxonomy, taxonomy_name: str, branch_name
         self._create_child_links(taxonomy.child_links, project_label)
         self._create_previous_links(taxonomy.previous_links, project_label)
 
-    def __call__(self, filename: str, branch_name: str, taxonomy_name: str):
+    def __call__(
+        self,
+        main_filename: str,
+        external_filenames: list[str] | None,
+        branch_name: str,
+        taxonomy_name: str,
+    ):
         """Process the file"""
         start_time = timeit.default_timer()
 
         branch_name = normalize_text(branch_name, char="_")
         taxonomy_parser = TaxonomyParser()
         try:
-            taxonomy = taxonomy_parser.parse_file(filename, self.parser_logger)
+            if external_filenames is None:
+                taxonomy = taxonomy_parser.parse_file(main_filename, self.parser_logger)
```

Comment:
it's always better to have a code, that first unify cases, and then process.

I would rather write this as:
```python
            external_entry_nodes  = []
            if external_filenames:
                for filename in external_filenames:
                ... 
            taxonomy = taxonomy_parser.parse_file(
                    main_filename, self.parser_logger, external_entry_nodes
                )
```